### ISSUE TITLE
Allow overriding DBT version

### ISSUE THREAD

**Issue by ni-tami** [mage-ai/mage-ai#5469]
**Is your feature request related to a problem? Please describe.**
I've been using DBT in mage for running models. For an improvement, I'd like to use DBT v1.9 (now still beta) snapshot functionalities, but currently, mage uses v1.7.x.

**Describe the solution you'd like**
I hope mage allows overriding DBT version, so that I could use the beta version if that's possible.

**Describe alternatives you've considered**
I haven't tried any alternatives within the use of mage but considered not using mage for pipelines that require newer version of DBT.

**Additional context**
For my use, I'm trying a feature in DBT v1.9.0b1 that allows more customization in snapshots, i.e. renaming some dbt generated columns as described in this issue https://github.com/dbt-labs/dbt-core/issues/10185

Thank you

**Comment by wangxiaoyou1993** [comment]
Have you tried specifying your dbt package version in your requirements.txt and see if that works?

**Comment by ni-tami** [comment]
Yes, I've tried this but the version solving is failed. I think because the required version by mage is `==1.7.4`, isn't it?

I've also tried forking mage to try edit the requirements.txt, setup.py, and their related use in the Dockerfile, but no luck yet. I think it might be because there were typos too in my attempt, I'd take a look at it again soon to maybe try support `>= 1.7, < 1.10`.
